domain/src/main/java/com/thoughtworks/go/config/materials/git/GitMaterial.java

```diff
@@ -369,7 +241,7 @@ public String getTypeForDisplay() {
     }
 
     public String getBranch() {
-        return this.branch;
+        return this.refSpecOrBranch;
```

Comment:
Renamed only the field so it's clear what it really points to and how it should be interpreted/treated. Did not rename the getter for fear of changing serialization or something.